### PR TITLE
feat!: config is always visible on landing page

### LINKED ISSUE #486: Simplify configuration access on main page
Configuration toggle no longer serves a purpose, we don't have it on subdomains and it makes UX on main page bit clunky (https://github.com/ipfs/service-worker-gateway/issues/289).

### Proposal

Inline configuration on the main landing page.
It will make it easier for users of this tech demo to see what is possible, and also simplify the way we do E2E and manual tests.

> image

### PR DESCRIPTION
**Pull request by SgtPooki**
## Title

feat!: config is always visible on landing page

## Description

This change adds the config section to the main landing page (i.e. helper-ui.tsx) and removes the config gear icon in the header, and direct access to an isolated config page (unless on a subdomain)

Some heading text was also added to the config to make it more clear that it is a config section.

Fixes https://github.com/ipfs/service-worker-gateway/issues/486

## Notes & open questions

We still have a page accessible at /#/ipfs-sw-config for subdomain users to be able to edit the config, and a test for that now.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -1,10 +1,11 @@
-import React from 'react'
-import { RouteContext } from '../context/router-context.jsx'
-import gearIcon from '../gear-icon.svg'
```
Comment (SgtPooki) [inline src/components/Header.tsx:3]: will delete this file. incoming update to PR

File: src/context/config-context.tsx [hunk 2]
```diff
@@ -1,21 +1,17 @@
 import React, { createContext, useCallback, useEffect, useState } from 'react'
 import { defaultDebug, defaultDnsJsonResolvers, defaultEnableGatewayProviders, defaultEnableRecursiveGateways, defaultEnableWebTransport, defaultEnableWss, defaultGateways, defaultRouters, getConfig, resetConfig, type ConfigDb } from '../lib/config-db.js'
-import { isConfigPage } from '../lib/is-config-page.js'
 import { getUiComponentLogger } from '../lib/logger.js'
 import type { ComponentLogger } from '@libp2p/logger'
 
-const isLoadedInIframe = window.self !== window.top
+// const isLoadedInIframe = window.self !== window.top
```
Comment (SgtPooki) [inline src/context/config-context.tsx]: will remove in incoming commit

File: src/lib/routing-render-checks.ts [hunk 3]
```diff
@@ -21,9 +21,10 @@ export async function shouldRenderRedirectPage (): Promise<boolean> {
 
 export async function shouldRenderConfigPage (): Promise<boolean> {
   const { isConfigPage } = await import('../lib/is-config-page.js')
+  const { isSubdomainGatewayRequest } = await import('./path-or-subdomain.js')
 
   const isRequestToViewConfigPage = isConfigPage(window.location.hash)
-  return isRequestToViewConfigPage
+  return isRequestToViewConfigPage && isSubdomainGatewayRequest(window.location)
```
Comment (SgtPooki) [inline src/lib/routing-render-checks.ts:27]: only render this individual config page on subdomain requests explicitly for `/#/ipfs-sw-config`

File: src/pages/helper-ui.tsx [hunk 4]
```diff
@@ -19,10 +21,20 @@ function HelperUi (): React.JSX.Element {
...
 
+  // if (isConfigPage(window.location.hash)) {
+  //   // when user on a subdomain explicitly requests /#/ipfs-sw-config
+  //   return (
+  //     <>
+  //       <Header />
+  //       <Config />
+  //     </>
+  //   )
+  // }
```
Comment (SgtPooki) [inline src/pages/helper-ui.tsx]: will remove in incoming commit

File: src/pages/helper-ui.tsx [hunk 5]
```diff
@@ -5,8 +5,10 @@ import Header from '../components/Header.jsx'
 import CidRenderer from '../components/input-validator.jsx'
 import { ConfigProvider } from '../context/config-context.jsx'
 import { ServiceWorkerProvider } from '../context/service-worker-context.jsx'
+// import { isConfigPage } from '../lib/is-config-page'
```
Comment (SgtPooki) [inline src/pages/helper-ui.tsx]: will remove

**Review by SgtPooki — COMMENTED** [review]
self review

**Review by SgtPooki — COMMENTED** [review]
self review after changes

File: src/pages/config.tsx [hunk 6]
```diff
@@ -75,7 +75,11 @@ const dnsJsonValidationFn = (value: string): Error | null => {
   }
 }
 
-function ConfigPage (): React.JSX.Element | null {
+export interface ConfigPageProps extends React.HTMLProps<HTMLElement> {
+
+}
+
+const ConfigPage: FunctionComponent<ConfigPageProps> = () => {
```
Comment (2color) [inline src/pages/config.tsx:83]:
```suggestion
// Config Page can be loaded either as a page or as a component in the landing helper-ui page
const ConfigPage: FunctionComponent<ConfigPageProps> = () => {
```

**Review by 2color — APPROVED** [review]
Nicer addition. 
LGTM. Also tested locally.

**Pull request merged**